assignments/assignments_parser_test.go

```diff
@@ -216,6 +221,93 @@ func TestParseUnknownFields(t *testing.T) {
 	}
 }
 
+func TestParseAndSaveAssignment(t *testing.T) {
+	testsDir := t.TempDir()
+
+	for _, c := range []struct {
+		path, filename, content string
+	}{
+		{"lab1", "assignment.yml", y1},
+		{"lab2", "assignment.yml", y2},
+	} {
+		writeFile(t, testsDir, c.path, c.filename, c.content)
+	}
+	course := &qf.Course{
+		ID:   1,
+		Name: "Test course",
+	}
+
+	db, cleanup := qtest.TestDB(t)
+	defer cleanup()
+
+	admin := qtest.CreateNamedUser(t, db, 1, "test")
+	if err := db.CreateCourse(admin.ID, course); err != nil {
+		t.Fatal(err)
+	}
+
+	assignments, _, err := readTestsRepositoryContent(testsDir, course.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(assignments) != 2 {
+		t.Errorf("len(assignments) = %d, want %d", len(assignments), 1)
+	}
+
+	// Save the assignments to the database
+	if err := db.UpdateAssignments(assignments); err != nil {
+		t.Fatal(err)
+	}
+
+	// Read the assignments from the database
+	gotAssignments, err := db.GetAssignmentsByCourse(course.ID)
+
```

Comment:
Remove the blank line. We try to keep the error checked right after the line that returned the `err`. (This applies to several cases below as well.)

PS: My VSCode Go extension configuration automatically fixes these types of things for me via the `gofumpt` formatter.